blocks/block-controls/index.js

```diff
@@ -1,13 +1,25 @@
 /**
  * WordPress dependencies
  */
-import { Toolbar, Fill } from '@wordpress/components';
-
-export default function BlockControls( { controls, children } ) {
-	return (
-		<Fill name="Block.Toolbar">
-			<Toolbar controls={ controls } />
-			{ children }
-		</Fill>
-	);
-}
+import { createSlotFill, Toolbar } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ifBlockEditSelected } from '../block-edit/context';
+
+const Fill = createSlotFill( 'BlockControls' );
```

Comment:
I have an improvement for this one in another PR:
https://github.com/WordPress/gutenberg/pull/6086/files#diff-f55821c6d9b2038d9e462f87c1c9e0feR6

I will extract and update this PR if it is going to be ready to merge sooner.